lib/blacklight/configuration.rb

```diff
@@ -252,8 +252,16 @@ def deep_copy
       end
       alias_method :inheritable_copy, :deep_copy
     else
-      alias_method :deep_copy, :deep_dup
-      alias_method :inheritable_copy, :deep_dup
+      ##
+      # Rails 4.x provides `#deep_dup`, but it aggressively `#dup`'s class names
+      # too. These model names should not be `#dup`'ed or we might break ActiveModel::Naming.
```

Comment:
thanks for the comment, I get it now. Those aren't class names, technically, those are classes themselves, where dup'ing would make a new class.